plugins/Diagnostics/Diagnostic/DbMaxPacket.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Piwik\Plugins\Diagnostics\Diagnostic;
+
+use Piwik\Db;
+use Piwik\MetricsFormatter;
+use Piwik\Piwik;
+use Piwik\SettingsPiwik;
+use Piwik\Translation\Translator;
+
+/**
+ * Check if Piwik is connected with database through ssl.
+ */
+class DbMaxPacket implements Diagnostic
+{
+    /**
+     * @var Translator
+     */
+    private $translator;
+
+    const MIN_VALUE_MAX_PACKET_MB = 32;
+
+    public function __construct(Translator $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function execute()
+    {
+        if (!SettingsPiwik::isPiwikInstalled()) {
+            return array(); // only possible to perform check once we have DB connection
+        }
+
+        $maxPacketBytes = Db::fetchRow("SHOW VARIABLES LIKE 'max_allowed_packet'");
+
+        $status = DiagnosticResult::STATUS_OK;
+        $label = $this->translator->translate('Diagnostics_MysqlMaxPacketSize');
+        $comment = '';
+
+        $minSize = self::MIN_VALUE_MAX_PACKET_MB * 1000 * 1000; // not using 1024 just in case... this amount be good enough
+        if (!empty($maxPacketBytes['Value']) && $maxPacketBytes['Value'] < $minSize) {
+            $status = DiagnosticResult::STATUS_WARNING;
+            $pretty = MetricsFormatter::getPrettySizeFromBytes($maxPacketBytes['Value'], 'M');
+            $configured = str_replace(array(' M', '&nbsp;M'), 'MB', $pretty);
+            $comment = Piwik::translate('Diagnostics_MysqlMaxPacketSizeWarning', array('32MB', $configured));
```

Comment:
This should not be hardcoded I think.

Reply:
The referenced issue suggest to set it to at least 64MB, but guess 32MB might be fine as well?

Reply:
I don't remember exactly but we talked about it in chat or so. Somewhere 32 was said. Can change it to 64.